sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsSpanImpl.java

```diff
@@ -0,0 +1,510 @@
+  @GuardedBy("this")
+  @Nullable
+  private EvictingQueue<SpanData.Link> links;
+  // Number of link recorded.
+  @GuardedBy("this")
+  private int totalRecordedLinks = 0;
+  // The number of children.
+  @GuardedBy("this")
+  private int numberOfChildren;
+  // The status of the span.
+  @GuardedBy("this")
+  @Nullable
+  private Status status;
+  // The end time of the span.
+  @GuardedBy("this")
+  private long endNanoTime;
+  // True if the span is ended.
+  @GuardedBy("this")
+  private boolean hasBeenEnded;
+
+  /**
+   * Creates and starts a span with the given configuration.
+   *
+   * @param context supplies the trace_id and span_id for the newly started span.
+   * @param name the displayed name for the new span.
+   * @param parentSpanId the span_id of the parent span, or null if the new span is a root span.
+   * @param traceConfig trace parameters like sampler and probability.
+   * @param spanProcessor handler called when the span starts and ends.
+   * @param timestampConverter null if the span is a root span or the parent is not sampled. If the
+   *     parent is sampled, we should use the same converter to ensure ordering between tracing
+   *     events.
+   * @param clock the clock used to get the time.
+   * @return a new and started span.
+   */
+  @VisibleForTesting
+  public static RecordEventsSpanImpl startSpan(
+      SpanContext context,
+      String name,
+      Kind kind,
+      @Nullable SpanId parentSpanId,
+      TraceConfig traceConfig,
+      SpanProcessor spanProcessor,
+      @Nullable TimestampConverter timestampConverter,
+      Clock clock,
+      Resource resource) {
+    RecordEventsSpanImpl span =
+        new RecordEventsSpanImpl(
+            context,
+            name,
+            kind,
+            parentSpanId,
+            traceConfig,
+            spanProcessor,
+            timestampConverter,
+            clock,
+            resource);
+    // Call onStart here instead of calling in the constructor to make sure the span is completely
+    // initialized.
+    spanProcessor.onStartSync(span);
+    return span;
+  }
+
+  /**
+   * Returns the name of the {@code Span}.
+   *
+   * @return the name of the {@code Span}.
+   */
+  @Override
+  public String getName() {
+    synchronized (this) {
+      return name;
+    }
+  }
+
+  @Override
+  public io.opentelemetry.proto.trace.v1.Span toSpanProto() {
+    throw new UnsupportedOperationException("to be implemented");
+  }
+
+  /**
+   * Returns the status of the {@code Span}. If not set defaults to {@link Status#OK}.
+   *
+   * @return the status of the {@code Span}.
+   */
+  public Status getStatus() {
+    synchronized (this) {
+      return getStatusWithDefault();
+    }
+  }
+
+  /**
+   * Returns the end nano time (see {@link System#nanoTime()}). If the current {@code Span} is not
+   * ended then returns {@link Clock#nowNanos()}.
+   *
+   * @return the end nano time.
+   */
+  public long getEndNanoTime() {
```

Comment:
Why public? Same question for every method that is not an override?

Reply:
Updated.